src/View/ViewableData.php

```diff
@@ -562,9 +567,18 @@ public function obj($fieldName, $arguments = [], $cache = false, $cacheName = nu
         if (!is_object($value)) {
             // Force cast
             $castingHelper = $this->castingHelper($fieldName);
-            $valueObject = Injector::inst()->create($castingHelper, $fieldName);
-            $valueObject->setValue($value, $this);
-            $value = $valueObject;
+            // Cast arrays to ArrayList by default
+            if (is_array($value) && $castingHelper === $this->config()->get('default_cast')) {
+                $value = ArrayList::create($value);
+            } else {
```

Comment:
`ArrayList` converts associative arrays to `ArrayData` - should we do the same here?